config/src/main/java/com/typesafe/config/impl/ConfigNodeKey.java

```diff
@@ -0,0 +1,21 @@
+/**
+ *   Copyright (C) 2015 Typesafe Inc. <http://typesafe.com>
+ */
+package com.typesafe.config.impl;
+
+import java.util.Collection;
+
+final class ConfigNodeKey extends AbstractConfigNode {
```

Comment:
Sounds good. I think it would also be worth changing `ConfigNodeKeyValue` to `ConfigNodePathValue` so I'll go ahead and do that too.